RecoLocalCalo/HcalRecProducers/src/HcalHitSelection.cc

```diff
@@ -191,6 +190,8 @@ void HcalHitSelection::produce(edm::Event& iEvent, const edm::EventSetup& iSetup
   skim(hf, *hf_out);
   iEvent.put(std::move(hf_out), hfTag.label());
 
+  if (!iEvent.getByToken(tok_ho_, ho))
+    return;
   auto ho_out = std::make_unique<HORecHitCollection>();
   skim(ho, *ho_out, hoSeverityLevel);
   iEvent.put(std::move(ho_out), hoTag.label());
```

Comment:
in reco we should still put a declared product into the event (it can be empty) instead of returning.
The if -> return yesterday is more likely an indication of a problem or a bug in configuration.
Why is it done?

Reply:
reducedHcalRecHits as an input for slimmedHcalRecHits exists only for HBHE and HF collections in AOD, 
so some protection is needed here to re-use the same producer in the absence of HO reducedHcalRecHits
(with empty InputTag for HO input in cloned config for simplicity) for slimmedHcalRecHits output when 
there is re-miniAOD from AOD,  as in wf 136.88811_RunJetHT2018D.
Please, suggest an alternative.

Reply:
>  Please, suggest an alternative.

configure properly, so that the code does not try to read a collection that is known at configuration step to not exist

Reply:
Slava
(1) it works very well for all wf's as is 
(2) your suggestion sounds quite general/vague to me, could you elaborate a bit (given that the config is this PR) ? - 
I can think of adding a check in the code  if InputTag for HO is empty 
https://github.com/cms-sw/cmssw/blob/cb6479cf8032e9eda455b3870b510b204969de21/RecoLocalCalo/HcalRecProducers/src/HcalHitSelection.cc#L123
don't do anything for HO in (downstream) but it would imply a presence of several switches "if (noHO)" and doesn't look nice...

Reply:
a check for an empty InputTag or a separate flag should be fine.

Reply:
Sorry for confusion... :( 
Slava, you're right, while my previous comment was incorrect - there is always HO reducedHcalRecHits in AOD (albeit empty), so no need to add an empty tag to slimmedHcalRecHits producer (clone of reducedHcalRecHits) and to play around it. The simplest option should work.